File: example/java-example/src/main/java/com/java/example/model/BaseResponseMapper.java
package com.java.example.model;


import com.java.example.rest.model.BaseResponseRest;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface BaseResponseMapper {

    BaseResponse parse(BaseResponseRest input);

    BaseResponseRest parse(BaseResponse input);
}

File: example/webflux-example/src/main/kotlin/com/webflux/example/model/BaseResponseMapper.kt
package com.webflux.example.model

import com.webflux.example.rest.model.BaseResponseRest
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface BaseResponseMapper {
    fun parse(input: BaseResponseRest): BaseResponse
    fun parse(input: BaseResponse): BaseResponseRest
}
